easysocial/easysocial/album.php

```diff
@@ -42,38 +42,37 @@ public function get()
 
 	public function post()
 	{
-		$this->create_album();
+		$this->createAlbum();
 	}
 
 	/** DELETE Call
 	 * 
-	 * @return	array	message
+	 * @return  mixed
 	 */
 	public function delete()
 	{
-		return $this->plugin->setResponse($this->delete_check());
+		return $this->deleteCheck();
 	}
 
 	/** switch case for photo delete or album delete. 
 	 * 
-	 * @return	object
+	 * @return  ApiPlugin response object
 	 */
-	public function delete_check()
+	private function deleteCheck()
 	{
 		$app	=	JFactory::getApplication();
 		$flag	=	$app->input->get('flag', null, 'STRING');
```

Comment:
STRING filter is too permissive. It will allow any string. In this case CMD is best suited.

Reply:
Read about all available filters here - https://docs.joomla.org/Retrieving_request_data_using_JInput